### PR TITLE
Feature/nup 3858 ounat read template folder from provided directory path

### PR DESCRIPTION
**Pull request by osmanertem**
* add support for getting a template folder from an external path.
* add support for extending output project's dependencies(in package.json) with external template's dependencies. 
* update prompt.js to include questions for external template path.
* usage with `vue invoke `and  `vue create ... --preset`are tested

### CODE REVIEW

File: generator/index.js [hunk 1]
```diff
@@ -142,6 +142,13 @@ module.exports = async (api, options, rootOptions) => {
       fs.ensureFileSync(api.resolve(path.join(genConfig.dirPathPrefix, 'babel.config.js')));
       await applyBabelConfig(api, genConfig, 'babel.config.js');
     });
+
+    api.render(async () => {
+      if (options.templateType == 'custom') {
```
Comment (brendtumi) [inline generator/index.js]: I think better to check `options.customTemplatePath` is exist.

File: generator/index.js [hunk 2]
```diff
@@ -772,6 +779,47 @@ const getAllFilesInDirStructure = (module.exports.replaceInFile = async (srcPath
   }
 });
 
+// utility function used to extend package.json with the dependencies from custom template's package.json
+const extendPackageJSONWithCustomTemplatesDependencies = (module.exports.extendPackageJSONWithCustomTemplatesDependencies = async (api, srcPathPrefix) => {
+  try {
+    const packageJSONPath = path.join(srcPathPrefix, 'package.json');
+    if (!fs.existsSync(packageJSONPath)) {
+      return;
```
Comment (brendtumi) [inline generator/index.js:788]: We can return an informative console text about path not exist.

File: prompts.js [hunk 3]
```diff
@@ -62,8 +62,20 @@ Example: com.company.app`;
...
         name: 'Advanced',
         value: 'advanced'
+      },
+      {
+        name: 'Custom',
+        value: 'custom'
       }
     ],
-    default: 'simple'
+    default: 'custom'
```
Comment (mylmz10) [inline prompts.js]: I think the default value must be simple.

File: generator/index.js [hunk 4]
```diff
@@ -772,6 +779,47 @@ const getAllFilesInDirStructure = (module.exports.replaceInFile = async (srcPath
...
+    }
+
+    const templatesPackageJSON = fs.readJsonSync(packageJSONPath, { encoding: 'utf8' });
+
+    let newDeps = {};
```
Comment (brendtumi) [inline generator/index.js]: `const`. newDeps used once.

File: generator/index.js [hunk 5]
```diff
@@ -772,6 +779,47 @@ const getAllFilesInDirStructure = (module.exports.replaceInFile = async (srcPath
...
+    for (let currentDepName in templatesPackageJSON.dependencies) {
+      if (!api.hasPlugin(currentDepName)) {
+        newDeps[currentDepName] = templatesPackageJSON.dependencies[currentDepName];
+      }
+    }
+    api.extendPackage({ dependencies: newDeps });
+
+    let newDevDeps = {};
```
Comment (brendtumi) [inline generator/index.js]: `const` newDevDeps

**Review by brendtumi — CHANGES_REQUESTED** [review]

File: generator/index.js [hunk 6]
```diff
@@ -142,6 +142,13 @@ module.exports = async (api, options, rootOptions) => {
       fs.ensureFileSync(api.resolve(path.join(genConfig.dirPathPrefix, 'babel.config.js')));
       await applyBabelConfig(api, genConfig, 'babel.config.js');
     });
+
+    api.render(async () => {
```
Comment (mylmz10) [inline generator/index.js]: Can I use custom template without using babel?

Comment (osmanertem) [reply generator/index.js]: template copy code block is moved out of if babel

File: generator/index.js (hunk 1 above)
Comment (osmanertem) [reply generator/index.js]: done

File: generator/index.js (hunk 2 above)
Comment (osmanertem) [reply generator/index.js:788]: done

File: generator/index.js (hunk 4 above)
Comment (osmanertem) [reply generator/index.js]: done

File: generator/index.js (hunk 5 above)
Comment (osmanertem) [reply generator/index.js]: done

File: prompts.js (hunk 3 above)
Comment (osmanertem) [reply prompts.js]: done

**Review by mylmz10 — APPROVED** [review]

**Review by brendtumi — APPROVED** [review]

**Pull request merged**